index.ts

```diff
@@ -1,35 +1,52 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {
+    if (options !== undefined) {
+      if (options.data !== undefined) { this.push(options.data) }
+      this._keys = options.keys
+    }
+  }
+
+  get keys(): string[] | undefined {
+    return this._keys
+  }
 
   get length(): number {
     return this._original.length
   }
 
-  constructor(input: object[], keys: string[]) {
-    this._original = input.map((x) => Object.assign({}, x))
-    this._keys = keys
+  get original(): object[] {
+    return this._original
+  }
 
-    this._transposed = transpose(this._keys, this._original)
+  get transposed(): object[] {
+    return this._original
   }
 
   public get(keyName: string, keyValue: string): object[] {
-    return this._transposed[keyName][keyValue]
+    return this._transposed[keyName][keyValue] || []
   }
 
-  public push(...items: object[]): number {
-    this._transposed = transpose(this._keys, items, this._transposed)
-    return this._original.push(...items)
+  public push(data: object[]) {

```

Comment:
Rest/spread parameters are still new to me; I didn't mean to commit this. 
Are the before and after equivilent?

Reply:
`push`ing without spreading is different, e.g. given:

```ts
const values = [1, 2, 3]
const array = [0]

array.push(values) // [0, [1, 2, 3]]
// or
array.push(...values) // [0, 1, 2, 3]
```

Reply:
Ah okay, makes sense. Will revert. Spreading is clearly the more desirable option here

Reply:
yeah `push`ing an array just doesn't make sense